src/outpack/location.py

```diff
@@ -0,0 +1,108 @@
+import shutil
+
+from outpack.config import (
+    Location,
+    update_config,
+)
+from outpack.root import root_open
+from outpack.util import match_value
+
+LOCATION_LOCAL = "local"
+LOCATION_ORPHAN = "orphan"
+LOCATION_RESERVED_NAME = [LOCATION_LOCAL, LOCATION_ORPHAN]
+LOCATION_TYPES = [LOCATION_LOCAL, LOCATION_ORPHAN, "path"]
+
+
+def orderly_location_list(root=None, locate=True):
```

Comment:
Throughout, I think we should use `outpack_` prefixes until we sort out the naming (can discuss later today) or it will be even more complicated to think about.

To avoid the lint issue, this would look like

```suggestion
def orderly_location_list(*, root=None, locate=True):
```

and below

```
def orderly_location_add(name, type, args, *, root=None, locate=True):
```